subworkflows/nf-core/mapping.nf

```diff
@@ -36,27 +35,30 @@ workflow MAPPING {
         SAMTOOLS_STATS ( bam_sorted_indexed )
 
         // Merge multiple lane samples and index
-        SAMTOOLS_SORT.out.bam.map{ meta, bam ->
-            new_meta = meta.clone()
-            new_meta.id = new_meta.id.split('_')[0]
-            [new_meta, bam]
-        }.groupTuple().branch{
+        SAMTOOLS_SORT.out.bam
+        .map{ meta, bam ->
+            new_meta = meta.clone()                 // clone to avoid overriding the global meta
+            new_meta.id = new_meta.id.split('_')[0] // access the .id attribute of meta to split samplename_lane into samplename
+            [new_meta, bam]}                        // end the closure to return newly modified channel
+        .groupTuple(by: 0)                          // group them bam paths with the same [ samplename, [[bam path], [bam path], ..] ]
+        .branch{                                    // branch the channel into multiple channels (single, multiple) depending on size of list
             single: it[1].size() == 1
             multiple: it[1].size() > 1
-        }.set{ bams_to_merge }
+            }
+        .set{ bams }                                // create a new multi-channel named bams
```

Comment:
@jemten I changed the formatting + applied comments! 🤓 🎉

Reply:
Now it's a bit easier for me to understand 😄